lib/ecto.ex

```diff
@@ -595,4 +595,21 @@ defmodule Ecto do
       true
     end
   end
+
+  @spec load(Ecto.Schema.t | map(), map() | Keyword.t | {list, list}) :: Ecto.Schema.t | map()
```

Comment:
@josevalim I took a stab at defining a type spec for this. WDYT?

With latest changes we'd basically support 3 ways of passing data:

```elixir
Ecto.load(User, %{name: "joe"})
Ecto.load(User, [name: "joe"])
Ecto.load(User, {[:name], ["joe"]})
```

the 3rd one is very similar to @michalmuskala's original proposal to pass columns & values.

I finally realized that `Schema.__load__` was getting data primarly as `{fields, values}` (e.g. from DB), and `__load__` accepting a map was only used to fill in assoc/map fields.

One thing that is now changed is that `Ecto.load(User, %{"name" => "joe"})` is no longer supported out of the box (I could add key stringification to Ecto.load) and I think it's actually for the better - if it's supposed to be fairly low-level method then I think it's fine to assume that field names are atoms. (again, when `Ecto.__load__` is called by the code that grabbed data from DB, column names were already converted to atoms)